packages/eas-cli/src/metadata/config.ts

```diff
@@ -12,6 +18,83 @@ export interface MetadataConfig {
   apple?: AppleMetadata;
 }
 
+/**
+ * Resolve the dynamic config from the user.
+ * It supports methods, async methods, or objects (json).
+ */
+async function resolveDynamicConfigAsync(configFile: string): Promise<unknown> {
+  const userConfigContent = await import(configFile);
+  let userConfig = userConfigContent.default || userConfigContent;
+
+  if (typeof userConfig === 'function') {
+    userConfig = await userConfig();
+  }
+
+  if (typeof userConfig === 'object' && userConfig !== null && !Array.isArray(userConfig)) {
```

Comment:
The next call after this function will be validation, and by returning unknown you are losing the information you are checking here.

Either run full validation as part of this function or do not validate at all.

Reply:
Done, validation is now moved to `loadConfigAsync`, and `skipValidation` option is added ([just like the CLI options](https://github.com/expo/expo/search?l=TypeScript&q=skip)).